### PR TITLE
options.json increased supported parameters

### PR DESCRIPTION
**Pull request by GeoDerp**
Aligned Parameter list to try and match list in config_emhass.yaml .
The hope is to create a future commit to the emhass-add-on repository  to support said parameters.

### CODE REVIEW

**Comment by GeoDerp** [comment]
Will have a look at the array keys and tests tomorrow *(when I'm less tired)*.
May look into merging the methods/changes seen in https://github.com/davidusb-geek/emhass/pull/69 to assist.

**Comment by GeoDerp** [comment]
Once @davidusb-geek is happy, and the commits have been pulled. I can start looking into merging the other *(still relevant)* commits from @smurfix pull request https://github.com/davidusb-geek/emhass/pull/69

**Comment by GeoDerp** [comment]
Found a few bugs with this pull request. Fill try and fix them soon

**Comment by davidusb-geek** [comment]
Ok thank you for this work. Integrating those improvements from @smurfix was a long standing item in the to do list.
They are very tricky so we really need to test this very well.
Keep in mind that the file `web_server.py` is not tracker by the tests because I have not found a proper way to unit test a web sever like this. So any changes that may affect the web server need to be tested very well. Chaning that yaml file configuration list thing may affect the web server, so be carefull about that.
I have some comments on the reviews section

File: src/emhass/utils.py [hunk 1]
```diff
@@ -391,7 +391,7 @@ def get_yaml_parse(config_path: str, use_secrets: Optional[bool] = True,
         else:
             input_secrets = input_conf.pop('params_secrets', None)
         
-    retrieve_hass_conf = dict((key,d[key]) for d in input_conf['retrieve_hass_conf'] for key in d)
+    retrieve_hass_conf = input_conf['retrieve_hass_conf']
```
Comment (davidusb-geek) [inline src/emhass/utils.py:394]: Why did you change these, we are just integrating the dicts

File: src/emhass/web_server.py [hunk 2]
```diff
@@ -253,36 +267,50 @@ def action_call(action_name):
...
+            options['list_surface_azimuth'] = []
+            options['list_modules_per_string'] = []
+            options['list_strings_per_inverter'] = []
+
         DATA_PATH = "/share/" #"/data/"
     else:
         CONFIG_PATH = os.getenv("CONFIG_PATH", default="/app/config_emhass.yaml")
-        options = None
-        DATA_PATH = os.getenv("DATA_PATH", default="/app/data/")
+        options = {}
```
Comment (davidusb-geek) [inline src/emhass/web_server.py:300]: Again careful with these changes.
What is the need behind these changes?

File: src/emhass/web_server.py [hunk 3]
```diff
@@ -297,18 +325,18 @@ def action_call(action_name):
     else:
         injection_dict = None
     
-    if args.addon == 1:
+    if hass_url:
```
Comment (davidusb-geek) [inline src/emhass/web_server.py:328]: The web server needs to be generic for the add-on but also for the standalone docker installation method. That argument `args.addon` was a way to know which installation method was used and how we access the configuration file.

**Comment by GeoDerp** [comment]
haha yeah these are pretty mutch all the issues i have been finding.

**Comment by davidusb-geek** [comment]
Hi @GeoDerp, as you can see the main branch as been updated following the merge of #153.
Unfortunately there are some conflicts because @michaelpiron also changed the old list behavior in the config file.
Please update your branch and solve the eventual conflicts. Thank you

**Comment by davidusb-geek** [comment]
@GeoDerp what is the current status of this? Did you managed to test the web_server.py?

**Comment by GeoDerp** [comment]
Haven't made any new tests, but from my local tests everything seems functional. Just need to find a way to parse options into local builds to be sure. Haven't worked out how that works yet.

**Comment by davidusb-geek** [comment]
Maybe we can add an option `OPTIONS_PATH` to the web server for the case where we are not using the add-on and then pass the `options.json` when executing the `docker run` command. But make this optional.

**Comment by davidusb-geek** [comment]
Ok I will merge and try to test it locally on my side.

**Review by davidusb-geek — APPROVED** [review]
